### PR TITLE
Feature/transcribe video files

### PR DESCRIPTION
**Pull request by turicas**
Hi! I found out the `audiotools/speech.py` script (used via CLI) can transcribe many more audio formats and video files too. So I was wondering: why not support these formats on TranscriberBot? It could be very useful in many cases, like:

- When one forwards audio from WhatsApp to Telegram directly to TranscriberBot, the file format could be in `.ogx` or `.ogg` extension. The bot currently ignores `.ogx` files;
- Creating subtitles for videos would be easier if we can just send the video file to the bot instead of first converting them to audio (this is specially useful if we're with our cellphone and unable to convert quickly). For many use cases the video quality is low and the audio quality is ok, like when recording in WhatsApp or Telegram directly (so the files aren't big).

The code already supports many more extensions then the default set, but the bot ignores it. So, why not?

To make the tests I've used [obs](https://obsproject.com/) to record a video with audio (saved as `testdata/video.mkv`). Then, I converted the video to many other video and audio formats using [ffmpeg](https://ffmpeg.org/):

```shell
ffmpeg -i testdata/video.mkv testdata/video.avi  # video to video
ffmpeg -i testdata/video.mkv testdata/video.mp3  # video to audio
ffmpeg -i testdata/video.mkv testdata/video.ogg  # video to video (!)
ffmpeg -i testdata/video.mkv testdata/video.ogv  # video to video
ffmpeg -i testdata/video.mkv testdata/video.webm # video to video
ffmpeg -i testdata/video.mp3 testdata/video-audio-only.ogg  # audio to audio (!)
```

The file sizes differ a lot, but the audio "content" is the same for all of them.

> (!) Note: Telegram uses `ogg` as audio extension, but the file format could handle video also. In the example above I've created two ogg files: one with video + audio (called `video.ogg`) and another with audio only (called `video-audio-only.ogg`). The second one was converted from the MP3 file, which is audio-only.

Then, I've set `WIT_API_KEY` environment variable and run the following commands to try to transcribe all the files:

```shell
export TIMEFORMAT="%R"
for filename in testdata/video*; do
	echo "Transcribing $filename ($(du -h $filename | cut -f 1)):"
	time python src/audiotools/speech.py $WIT_API_KEY $filename -
	echo
done
```

The result was:

```text
Transcribing testdata/video-audio-only.ogg (84K):
Olá, esse é um teste de áudio e vídeo. Vamos testar o é isso aí.
2.660

Transcribing testdata/video.avi (3.4M):
Olá, esse é um teste de áudio e vídeo. Vamos testar o é isso aí.
2.819

Transcribing testdata/video.mkv (8.3M):
Olá, esse é um teste de áudio e vídeo. Vamos testar o é isso aí.
2.729

Transcribing testdata/video.mp3 (180K):
Olá, esse é um teste de áudio e vídeo. Vamos testar o é isso aí.
2.596

Transcribing testdata/video.mp4 (1.6M):
Olá, esse é um teste de áudio e vídeo. Vamos testar o é isso aí.
3.020

Transcribing testdata/video.ogg (740K):
Olá, esse é um teste de áudio e vídeo. Vamos testar o é isso aí.
2.660

Transcribing testdata/video.ogv (740K):
Olá, esse é um teste de áudio e vídeo. Vamos testar o é isso aí.
2.861

Transcribing testdata/video.webm (1.1M):
Olá, esse é um teste de áudio e vídeo. Vamos testar o é isso aí.
2.654
```

So the CLI was able to transcribe all the files correctly (with the same results), with some time differences. The size/durations:

| Format | Contents        | Original Size  | Duration |
|--------|-----------------|----------------|----------|
| ogg    | audio only      |           84KB |   2.660s |
| avi    | audio and video |          3.4MB |   2.819s |
| mkv    | audio and video |          8.3MB |   2.596s |
| mp3    | audio only      |          180KB |   2.596s |
| mp4    | audio and video |          1.6MB |   3.020s |
| ogg    | audio and video |          740KB |   2.660s |
| ogv    | audio and video |          740KB |   2.861s |
| webm   | audio and video |          1.1MB |   2.654s |

Duration average: 2.733s, stdev: 0.151s. The duration is more or less the same even if the original file sizes differ, since `pydub` extracts the audio and convert it to raw before sending to [wit.ai](https://wit.ai/)'s API (in the cases above, the raw audio sent was ~180KB, even for the first case, where the original file is smaller than it). So the only bad thing about transcribing videos is to download (and temporarly store) a bigger file than usual, but the traffic to Wit is more or less the same as if the original file was an audio-only file.

> Note: I've added only more used formats to implement the use cases I described (like forwarding the audio/video from WhatsApp or recording directly in Telegram), but the code may support [many other video formats](https://en.wikipedia.org/wiki/Video_file_format), so feel free to add more.

### CODE REVIEW

**Comment by stefanodelbosco** [comment]
Hi,
when we started with TranscriberBot, the first Whatsapp audio format was `.ogg`.

Before TranscriberBot become opensource, we had tried to add multiple audio format support.
There are some problems, with some format for example `.wav`, `.mp3` and `.flac`. 
The main problem is the 70-80% of audio received (with the format `.mp3` and `.flac`) were music. 
The bot took a long time to try to transcribe these audios and the result it was some delay between to send and receive a response.
The most senders were the groups and channels.

At the moment the only video supported is the 'video note', because it is similar an voice message (.mp4 with maximum duration one minute)

I view your commits and I saw you had added the support for `.avi`, `.mkv`, `mp4` (some GIF in Telegram are `.mp4`)
I think it will better add the video support only in private chats and maybe add other audio format (as `.mp3` and `.flac`) only in private chats.

Because, a lot of chats using TranscriberBot, in my opinion it is better to limit the transcription of messages to only 'valid' audios.

For example, I send a lot of GIF/short videos in some groups (where TranscriberBot is a member) and I would not like to have the various transcripts of those videos. 
The various transcripts these would became 'spam' messages and some groups would remove the bot.

I talked with @carloalbertobarbano and he agrees.

**Review by stefanodelbosco — CHANGES_REQUESTED** [review]
I think it will better add the video support only in private chats and maybe add other audio format (as `.mp3` and `.flac`) only in private chats.

**Comment by turicas** [comment]
> I think it will better add the video support only in private chats and maybe add other audio format (as `.mp3` and `.flac`) only in private chats.

I agree! When I implemented this I didn't want to also translate on groups, just didn't know the robot was used in groups. I don't have a deep understanding on how the flow of messages/checks/permissions is implemented. Could you please help me on this? Maybe adding `if chat_id < 0: return` inside `process_media_voice` will work, but I don't know if video files will call this function too.

File: src/transcriberbot/handlers_messages.py [hunk 1]
```diff
@@ -230,8 +230,11 @@ def document(bot, update):
   m = update.message or update.channel_post
   file_name = m.document.file_name
   _, file_ext = os.path.splitext(file_name)
+  file_extension = file_ext[1:]
+  supported_audio_extensions = config.get_config_prop("app").get("audio_ext", [])
+  supported_video_extensions = config.get_config_prop("app").get("video_ext", [])
 
-  if file_ext[1:] not in config.get_config_prop("app")["audio_ext"]:
+  if file_extension not in supported_audio_extensions + supported_video_extensions:
```
Comment (carloalbertobarbano) [inline src/transcriberbot/handlers_messages.py:240]: I think something like this should work (just a rough idea)

```python
if file_extension not in supported_audio_extensions + supported_video_extensions:
    logger.info('extension %s not recognized', file_ext)
    return

if file_extension in supported_video_extensions and chat_id < 0:
   # discard videos in groups
   return
```

**Review by carloalbertobarbano — COMMENTED** [review]
I think the filter should be applied in the document handler

**Comment by turicas** [comment]
Done! Also added the max file size as a config instead of hard-coded value.

> Note: a great future refactor will be centralizing all settings-related functions in a `settings.py` so we avoid coupling and the code from many modules (such as `resources`) could be more isolated.

**Pull request merged**